core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -127,8 +147,29 @@ protected void validate(TableMetadata base) {
       }
     }
 
-    if (conflictDetectionFilter != null && base.currentSnapshot() != null) {
-      validateAddedDataFiles(base, startingSnapshotId, conflictDetectionFilter, caseSensitive);
+
+    if (validateNewDataFiles && base.currentSnapshot() != null) {
+      validateAddedDataFiles(base, startingSnapshotId, conflictDetectionFilter(), caseSensitive);
+    }
+
+    if (validateNewDeleteFiles && base.currentSnapshot() != null) {
+      if (rowFilter() != Expressions.alwaysFalse()) {
+        validateNoNewDeleteFiles(base, startingSnapshotId, conflictDetectionFilter(), caseSensitive);
```

Comment:
Takes a bit of time to understand this if - else, especially as conflictDetectionFilter() itself returns conditionally.  Wonder if it can be simplified any way. 

Tracing through the many cases, some cases seem weird, eg:

If (rowFilter != false && !deletedDataFiles.isEmpty && conflictDetectionFilter == true), seems we will call validateNoNewDeleteFiles with conflictDetectionFilter, is it intended?

Reply:
Changed the logic to behave a bit differently now.

Reply:
Great, this looks a lot better and easier to understand now